trytond/trytond/cache.py

```diff
@@ -291,6 +291,10 @@ def commit(cls, transaction):
                     cursor.execute(
                         'NOTIFY "%s", %%s' % cls._channel,
                         (json.dumps(list(sub_reset), separators=(',', ':')),))
+                for name in reset:
+                    if name in cls._instances:
+                        inst = cls._instances[name]
+                        inst._clear(dbname)
```

Comment:
What is the rationale for doing this before the actual commit in the database rather than after? I feel that doing it after would be better, but I am not completely sure of it.